testproject/Assets/GlobalGameState/Scripts/InGameManager.cs

```diff
@@ -0,0 +1,442 @@
+
+    //The local player
+    private NetworkedObject m_LocalPlayerObject;
+
+    private void Awake()
+    {
+#if UNITY_EDITOR
+        if ( NetworkingManager.Singleton == null)
+        {
+            GlobalGameState.s_EditorLaunchingAsHost = m_LaunchAsHostInEditor;
+            //This will automatically launch the MLAPIBootStrap and then transition directly to the scene this control is contained within (for easy development of scenes)
+            GlobalGameState.LoadBootStrapScene();
+            return;
+        }
+#endif
+        if (NetworkingManager.Singleton.IsListening)
+        {
+            if (IsServer)
+            {
+                GlobalGameState.Singleton.allPlayersLoadedScene += AllPlayersLoadedScene;
+                NetworkingManager.Singleton.OnClientDisconnectCallback += OnClientDisconnectCallback;
+            }
+
+            m_InGameState.OnValueChanged += InGameStateValueChanged;
+        }
+    }
+
+    /// <summary>
+    /// Start
+    /// Handle the initialization of dialog (text) and registering callbacks
+    /// </summary>
+    void Start()
+    {
+        if (m_WaitingForPlayers)
+        {
+            m_WaitingForPlayers.enabled = true;
+        }
+
+        if (m_GameIsPaused)
+        {
+            m_GameIsPaused.enabled = false;
+        }
+
+        if (m_ExitingGame)
+        {
+            m_ExitingGame.enabled = false;
+        }
+
+        if (!IsServer && m_HostInfo)
+        {
+            m_HostInfo.enabled = false;
+        }
+
+        SetAndInitializeLocalPlayer();
+        FreezeAndShowPlayers();
+    }
+
+    /// <summary>
+    /// SetAndInitializeLocalPlayer
+    /// Sets our local player, assigns the local main camera instanve to the local player, and
+    /// registers all NetworkedObjects with a RandomPlayerMover component for In-Game state changes
+    /// </summary>
+    void SetAndInitializeLocalPlayer()
+    {
+        NetworkedObject[] NetoworkedObjects = GameObject.FindObjectsOfType<NetworkedObject>();
+        foreach(NetworkedObject networkedObject in NetoworkedObjects)
+        {
+            RandomPlayerMover PlayerMover = networkedObject.GetComponent<RandomPlayerMover>();
+
+            if (PlayerMover)
+            {
+                if (networkedObject.IsLocalPlayer)
+                {
+                    m_LocalPlayerObject = networkedObject;
+                    PlayerMover.SetPlayerCamera();
+                }
+                PlayerMover.OnRegisterInGameManager(this);
+            }
+        }
+    }
+
+    /// <summary>
+    /// FreezeAndShowPlayers
+    /// This is to handle pausing and unpausing primarily the players' networked object clones not owned by the local player
+    /// </summary>
+    /// <param name="shouldFreeze">should we freeze everyone or un-freeze them?</param>
+    void FreezeAndShowPlayers(bool shouldFreeze = false)
+    {
+        NetworkedObject[] NetoworkedObjects = GameObject.FindObjectsOfType<NetworkedObject>();
```

Comment:
SpawnManager.SpawnedObjects could be used instead to show best practices/ better performance maybe?

Reply:
Very nice catch Luke!
Yeah, that is a much better approach a provides yet another use case scenario of MLAPI features.
As opposed to parsing through all GameObjects looking for the existing NetworkedObjects, just use the existing SpawnedObjects within the SpawnManager.  (your suggestion has been applied!)
TY!  :)